payments/providers/bambora_payform.py

```diff
@@ -97,7 +119,15 @@ def handle_initiate_payment(self, order: Order, response) -> str:
         if result == 0:
             # Create the URL where user is redirected to complete the payment
             # Append "?minified" to get a stripped version of the payment page
-            return self.url_payment_token.format(token=response["token"])
+            token = response["token"]
+
+            if order_token := (
+                OrderToken.objects.filter(order=order).order_by("created_at").first()
```

Comment:
Would it make sense to pass the created `order_token.id` to this function and update based on id here?

Reply:
That would be an alternative. I tried to keep the function signatures the same (to try to decouple the `BaseProvider` with the `OrderToken` flow) to properly implement it, but we could also add it and save one call to the db 🤔

Reply:
I think there is no reason to try to hide the dependency to `OrderToken`, instead we should make it more visible to make it easier to understand and follow.

Reply:
🤔 alrighty then